api/unstable/ktls.h

```diff
@@ -94,6 +95,56 @@ S2N_API int s2n_connection_ktls_enable_send(struct s2n_connection *conn);
  */
 S2N_API int s2n_connection_ktls_enable_recv(struct s2n_connection *conn);
 
+/**
+ * Allows kTLS to be enabled if a connection negotiates TLS1.3.
+ *
+ * This method must be called BEFORE enabling ktls on a connection using
+ * s2n_connection_ktls_enable_send or s2n_connection_ktls_enable_recv.
+ *
+ * Currently, the kernel does not support updating connection keys. However, the
+ * TLS1.3 protocol requires implementations update their sending or receiving keys
+ * when required by cryptographic key usage limits or when requested by a peer.
+ * A fix for the kernel is being worked on, but is not yet available.
+ *
+ * Therefore to enable TLS1.3 with kTLS, the following requirements must be met:
+ *
+ * If kTLS is enabled for receiving, the peer must NOT trigger a key update by
```

Comment:
I think these sections are way too long. I would start out bluntly with your actual instructions to the user: "An application _should_ control both clients and servers involved in TLS connections before attempting TLS13 kTLS connections. This is because...."

I feel like right now your approach is to go for "as technically accurate as possible" but I just don't think anyone is going to actually read all this text. IMO it's better to be explicit in the first sentence and add detail later for readers who want to go in depth. Maybe you could separate the requirements from the actual rational?

Reply:
You are 100% right. I've rewritten to try to make it actually usable, even though it'll be less informative :)